pyprocessing/renderer/tkinter/window.py

```diff
@@ -98,34 +117,55 @@ def set_polygon(self, points):
         action = Action(
             self.canvas, 'create_polygon',
             points, fill=self.pp.namespace.fill.hex,
-            outline=self.pp.namespace.stroke.hex,
+            outline=None,
+            width=self.pp.namespace.stroke_thickness,
+        )
+        # Makes the corners with the right shape
+        line_action = Action(
+            self.canvas, 'create_line',
+            *points, points[0], points[1],
+            fill=self.pp.namespace.stroke.hex,
+            joinstyle=self.pp.namespace.join,
+            width=self.pp.namespace.thickness,
         )
         self.queued_actions.append(action)
+        self.queued_actions.append(line_action)
 
     def set_arc(self, x, y, width, height, start, stop, mode):
         x1 = x - (width + 1) // 2
         y1 = y - (height + 1) // 2
         x2 = x + width // 2 + 1
         y2 = y + height // 2 + 1
-        start = degrees(start)
-        stop = degrees(stop)
-        angle = start - stop
+        start_degree = degrees(start)
+        stop_degree = degrees(stop)
+        angle = start_degree - stop_degree
         mode = mode.lower()
         if mode == 'pie':
             mode += 'slice'
         action = Action(
             self.canvas, 'create_arc',
             x1, y1, x2, y2,
-            start=start, extent=angle, style=mode,
+            start=start_degree, extent=angle, style=mode,
             fill=self.pp.namespace.fill.hex,
             outline=self.pp.namespace.stroke.hex,
+            width=self.pp.namespace.stroke_thickness,
         )
         self.queued_actions.append(action)
 
     def set_point(self, x, y):
+        if self.pp.namespace.stroke_cap not in {'round', 'projecting'}:
```

Comment:
don't use a set. Use a tuple. It's cleaner to use immutable types in that instance